x/metadata/keeper/bank_test.go

```diff
@@ -0,0 +1,561 @@
+package keeper_test
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/suite"
+
+	"cosmossdk.io/collections"
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/provenance-io/provenance/app"
+	"github.com/provenance-io/provenance/testutil/assertions"
+	"github.com/provenance-io/provenance/testutil/testlog"
+	"github.com/provenance-io/provenance/x/metadata/keeper"
+	"github.com/provenance-io/provenance/x/metadata/types"
+)
+
+func TestBankTestSuite(t *testing.T) {
+	suite.Run(t, new(BankTestSuite))
+}
+
+type BankTestSuite struct {
+	suite.Suite
+
+	app *app.App
+	ctx sdk.Context
+	bk  *keeper.MDBankKeeper
+
+	logBuffer bytes.Buffer
+}
+
+func (s *BankTestSuite) SetupTest() {
+	// Swap in the buffered logger maker so it's used in app.Setup, but then put it back (since that's a global thing).
+	defer app.SetLoggerMaker(app.SetLoggerMaker(app.BufferedInfoLoggerMaker(&s.logBuffer)))
+	s.app = app.Setup(s.T())
+	s.logBuffer.Reset()
+	s.ctx = s.app.NewContext(false)
+	s.bk = keeper.NewMDBankKeeper(s.app.BankKeeper)
+}
+
+// getLogOutput gets the log buffer contents and logs that to the test.
+// The returned value is the contents split on newline with empty lines removed from the end.
+// This (probably) also clears out the log buffer.
+func (s *BankTestSuite) getLogOutput(msg string, args ...interface{}) []string {
+	logOutput := s.logBuffer.String()
+	if len(strings.TrimSpace(logOutput)) == 0 {
+		s.T().Logf(msg+" log output: <none>", args...)
+		return nil
+	}
+	s.T().Logf(msg+" log output:\n%s", append(args, logOutput)...)
+
+	rv := strings.Split(logOutput, "\n")
+	for len(rv) > 0 && len(rv[len(rv)-1]) == 0 {
+		rv = rv[:len(rv)-1]
+	}
+	if len(rv) == 0 {
+		return nil
+	}
+	return rv
+}
```

Comment:
The `msg` and `args` are provided to `getLogOutput` the same way they'd be provided to `Logf` (or `Printf` etc.); it's assumed that the `msg` has a placeholder for each of the args (if there are any).

In the first highlighted line, your suggestion doesn't work because `msg` might have placeholders that you're no longer providing the values for. The string appended to `msg` doesn't have any placeholders, so there's nothing extra to add to `args`, but the `args` still need to be provided to `Logf`.

With the second highlighted line, I wanted it to be a single log entry. When `t.Logf` is used, the filename and line number are added to the left of the first line and all subsequent lines add whitespace to keep it all in line. If I split the line into two `Logf` (or `Log`) calls, that lead gets included twice and I'd rather only have it once. Since `msg` has placeholders for each of the `args`, and I'm adding a placeholder to the end of the `msg`, I need to also add the value to the end of the `args`, which is how I've got it.